packages/Python/lldbsuite/test/lang/swift/deserialization_failure/TestSwiftDeserializationFailure.py

```diff
@@ -0,0 +1,54 @@
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbutil as lldbutil
+import unittest2
+
+
+class TestSwiftDeserializationFailure(TestBase):
+
+    mydir = TestBase.compute_mydir(__file__)
+
+    def setUp(self):
+        TestBase.setUp(self)
+
+    def prepare(self):
+        import shutil
+        copied_source = self.getBuildArtifact("main.swift")
+        shutil.copyfile(os.path.join("Inputs", "main.swift"), copied_source)
+        self.build()
+        os.unlink(copied_source)
+        os.unlink(self.getBuildArtifact("a.swiftmodule"))
+
+    def run_tests(self, target, process):
+        static_bkpt = target.BreakpointCreateByName('staticTypes')
+        dynamic_bkpt = target.BreakpointCreateByName('dynamicTypes')
+        generic_bkpt = target.BreakpointCreateByName('genericTypes')
+        lldbutil.continue_to_breakpoint(process, static_bkpt)
+        self.expect("fr var hello", substrs=["(String)", "world"])
+
+        # We should not be able to resolve the types defined in the module.
+        lldbutil.continue_to_breakpoint(process, dynamic_bkpt)
+        self.expect("fr var c", substrs=["<could not resolve type>"])
+
+        lldbutil.continue_to_breakpoint(process, generic_bkpt)
+        # FIXME: this is formatted incorrectly.
+        self.expect("fr var t", substrs=["(T)"]) #, "world"])
+
+    @skipUnlessDarwin
+    @skipIf(debug_info=no_match(["dwarf"]))
```

Comment:
Why is this limited to `dwarf` only?

Reply:
I was going to say because I'm manually deleting the .swiftmodule so the dsym case is identical, but for the case with the damaged module it would make sense to test both code paths. Will change this.